src/dymaptic.Blazor.GIS.API.Core/Components/Views/MapView.razor.cs

```diff
@@ -478,8 +491,18 @@ protected override async Task RenderView(bool forceRender = false)
             return;
         }
 
-        if (Rendering || (Map is null && WebMap is null) || ViewJsModule is null ||
-            string.IsNullOrWhiteSpace(ApiKey)) return;
+        if (Rendering || (Map is null && WebMap is null) || ViewJsModule is null) return;
+
+        if (string.IsNullOrWhiteSpace(ApiKey) && (AllowDefaultEsriLogin is null || !AllowDefaultEsriLogin.Value))
+        {
+            ErrorMessage = "No ArcGIS API Key Found. See UsingTheAPI.md for instructions on providing an API Key or suppressing this message.";
+#if DEBUG
```

Comment:
Should we really put the entire `if` block inside the `#if DEBUG`? Since all we are doing is the print, there's no point in even checking if we are not in `DEBUG`.